File: reporter-web-app/src/models/ScannerDetails.js
/*
 * Copyright (C) 2019-2020 HERE Europe B.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * License-Filename: LICENSE
 */

class ScannerDetails {
    constructor(obj) {
        this.name = '';
        this.version = '';
        this.configuration = '';

        if (obj instanceof Object) {
            if (obj.name) {
                this.name = obj.name;
            }

            if (obj.version) {
                this.version = obj.version;
            }

            if (obj.configuration) {
                this.configuration = obj.configuration;
            }
        }
    }
}

export default ScannerDetails;

File: model/src/main/kotlin/ScannerDetails.kt
/*
 * Copyright (C) 2017-2019 HERE Europe B.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * License-Filename: LICENSE
 */

package org.ossreviewtoolkit.model

/**
 * Details about the used source code scanner.
 */
data class ScannerDetails(
    /**
     * The name of the scanner.
     */
    val name: String,

    /**
     * The version of the scanner.
     */
    val version: String,

    /**
     * The configuration of the scanner, could be command line arguments for example.
     */
    val configuration: String
) {
    companion object {
        /**
         * A constant for a [ScannerDetails] where all properties are empty strings.
         */
        @JvmField
        val EMPTY = ScannerDetails(
            name = "",
            version = "",
            configuration = ""
        )
    }
}
